filecoin-proofs/src/api/post.rs

```diff
@@ -595,18 +598,18 @@ pub fn verify_winning_post<Tree: 'static + MerkleTreeTrait>(
     let pub_params: compound_proof::PublicParams<fallback::FallbackPoSt<Tree>> =
         fallback::FallbackPoStCompound::setup(&setup_params)?;
 
-    let pub_sectors: Vec<_> = replicas
-        .iter()
-        .map(|(sector_id, replica)| {
+    let mut pub_sectors = Vec::with_capacity(param_sector_count);
+    for _ in 0..param_sector_count {
+        for (sector_id, replica) in replicas.iter() {
             let comm_r = replica.safe_comm_r().with_context(|| {
-                format!("verify_window_post: safe_comm_r failed: {:?}", sector_id)
+                format!("verify_winning_post: safe_comm_r failed: {:?}", sector_id)
             })?;
-            Ok(fallback::PublicSector {
+            pub_sectors.push(fallback::PublicSector {
                 id: *sector_id,
                 comm_r,
-            })
-        })
-        .collect::<Result<_>>()?;
+            });
+        }
+    }
```

Comment:
We need to roll this back. Winning PoSt will not actually change its behavior. We need the challenges of Winning PoSt to be sensitive to order, because otherwise they would all be the same. This is because a single challenge appears repeatedly at different positions within the list of challenged sectors. Each instance needs fresh challenges.